File: src/main/java/ru/piko/pikopluginlib/Files/CustomConfigFile.java
package ru.piko.allpikoplugin.Files;

import org.bukkit.Bukkit;
import org.bukkit.configuration.file.FileConfiguration;
import org.bukkit.configuration.file.YamlConfiguration;
import ru.piko.allpikoplugin.Main;

import java.io.File;

public abstract class CustomConfigFile {

    protected File file;
    protected FileConfiguration fileConfiguration;

    public CustomConfigFile(String nameFile) {
        Main plugin = Main.getPlugin();
        file = new File(plugin.getDataFolder(), nameFile + ".yml");
        if (!plugin.getDataFolder().exists()) {
            plugin.getDataFolder().mkdir();
        }
        if (!file.exists()) {
            try {
                file.createNewFile();
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
        fileConfiguration = YamlConfiguration.loadConfiguration(file);
    }
    public void save() {
        try {
            fileConfiguration.save(file);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: src/main/java/ru/piko/pikopluginlib/Files/CustomConfigFile.kt
package ru.piko.pikopluginlib.Files

import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.configuration.file.YamlConfiguration
import java.io.File

abstract class CustomConfigFile(dataFolder: File, relativePath: String) {
	protected var file: File = File(dataFolder, "$relativePath.yml")
	protected lateinit var conf: FileConfiguration
	
	fun init() {
		if (!file.parentFile.exists()) {
			file.parentFile.mkdirs()
		}
		if (!file.exists()) {
			try {
				file.createNewFile()
				onCreate()
			} catch (e: Exception) {
				e.printStackTrace()
			}
		}
		conf = YamlConfiguration.loadConfiguration(file)
		onLoad(false)
	}
	
	fun save() {
		try {
			conf.save(file)
			onSave();
		} catch (e: Exception) {
			e.printStackTrace()
		}
	}
	
	fun reload() {
		conf = YamlConfiguration.loadConfiguration(file)
		onLoad(true)
	}
	
	open fun onSave() {}
	
	open fun onCreate() {}
	
	open fun onLoad(reload: Boolean) {}
	
}